pkg/tracing/tracer_otel.go

```diff
@@ -0,0 +1,177 @@
+
+	exporter, err := otelExporterFactory(ctx, config, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	tracerProvider := sdktrace.NewTracerProvider(
+		sdktrace.WithBatcher(exporter),
+		sdktrace.WithResource(resource.NewWithAttributes(
+			semconv.SchemaURL,
+			semconv.ServiceName(fmt.Sprintf("%v-%v-%v-%v", appId.Project, appId.Environment, appId.Family, appId.Application)),
+		)),
+		sdktrace.WithSampler(sdktrace.ParentBased(sdktrace.TraceIDRatioBased(settings.SamplingRatio))),
+		sdktrace.WithRawSpanLimits(sdktrace.SpanLimits{
+			AttributeValueLengthLimit:   settings.AttributeValueLengthLimit,
+			AttributeCountLimit:         settings.AttributeCountLimit,
+			EventCountLimit:             settings.EventCountLimit,
+			LinkCountLimit:              settings.LinkCountLimit,
+			AttributePerEventCountLimit: settings.AttributePerEventCountLimit,
+			AttributePerLinkCountLimit:  settings.AttributePerLinkCountLimit,
+		}),
+	)
+
+	otel.SetTracerProvider(tracerProvider)
+	otel.SetTextMapPropagator(propagation.TraceContext{})
+
+	tracer := otel.GetTracerProvider().Tracer(
+		instrumentationName,
+		trace.WithInstrumentationVersion(instrumentationVersion),
+		trace.WithSchemaURL(semconv.SchemaURL),
+	)
+
+	return NewOtelTracerWithInterfaces(appId, tracer), nil
+}
+
+func NewOtelTracerWithInterfaces(appId cfg.AppId, tracer trace.Tracer) *otelTracer {
+	return &otelTracer{
+		AppId:  appId,
+		tracer: tracer,
+	}
+}
+
+func (t *otelTracer) StartSubSpan(ctx context.Context, name string) (context.Context, Span) {
+	ctx, span := t.tracer.Start(ctx, name)
+
+	return newOtelSpan(ctx, span)
+}
+
+func (t *otelTracer) StartSpan(name string) (context.Context, Span) {
+	return t.StartSubSpan(context.Background(), name)
+}
+
+func (t *otelTracer) StartSpanFromContext(ctx context.Context, name string) (context.Context, Span) {
+	if parentSpan := GetSpanFromContext(ctx); parentSpan != nil {
+		parentTrace := parentSpan.GetTrace()
+		return t.spanFromTrace(ctx, parentTrace, name)
+	}
+
+	if ctxTrace := GetTraceFromContext(ctx); ctxTrace != nil {
+		return t.spanFromTrace(ctx, ctxTrace, name)
+	}
+
+	return t.StartSubSpan(ctx, name)
+}
+
+func (t *otelTracer) HttpHandler(h http.Handler) http.Handler {
+	name := fmt.Sprintf("%v-%v-%v-%v", t.Project, t.Environment, t.Family, t.Application)
+	handlerFunc := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		span := trace.SpanFromContext(r.Context())
+
+		ctx, _ = newOtelSpan(ctx, span)
+		r = r.WithContext(ctx)
+
+		h.ServeHTTP(w, r)
+	})
+
+	return otelhttp.NewHandler(handlerFunc, name)
+}
+
+func (t *otelTracer) HttpClient(baseClient *http.Client) *http.Client {
+	return &http.Client{
+		Transport:     otelhttp.NewTransport(baseClient.Transport),
+		CheckRedirect: baseClient.CheckRedirect,
+		Jar:           baseClient.Jar,
+		Timeout:       baseClient.Timeout,
+	}
+}
+
+// GrpcUnaryServerInterceptor we still need to use the UnaryServerInterceptor because to maintain
```

Comment:
This sentence reads broken

Reply:
Fixed it, basically I want to say here why we still use the deprecated function instead of the stas.Handler.